packages/eslint-plugin/src/rules/no-unsafe-enum-comparison.ts

```diff
@@ -56,56 +58,81 @@ export default util.createRule({
     const parserServices = util.getParserServices(context);
     const typeChecker = parserServices.program.getTypeChecker();
 
+    function isMismatchedComparison(
+      leftNode: TSESTree.Node,
+      rightNode: TSESTree.Node,
+    ): boolean {
+      const left = parserServices.getTypeAtLocation(node.left);
+      const right = parserServices.getTypeAtLocation(node.right);
+
+      // Allow comparisons that don't have anything to do with enums:
+      //
+      // ```ts
+      // 1 === 2;
+      // ```
+      const leftEnumTypes = getEnumTypes(typeChecker, left);
```

Comment:
josh do you know why this isn't symmetrical? it looks like a bug, so if it is intentional i think it should have a comment.

Reply:
Which josh? 😄 and, what do you mean by "symmetrical"?

Reply:
you =p

mmm i mean this code:

```ts
        const leftEnumTypes = getEnumTypes(typeChecker, left);
        const rightEnumTypes = new Set(getEnumTypes(typeChecker, right));
        if (leftEnumTypes.length === 0 && rightEnumTypes.size === 0) {
          return;
        }
```

why aren't we getting a set on both sides?
or using an array on both sides?
i there's a reason I think you should put a comment and explain what the code is doing and why
(or you can copy paste it here and i will put it into this pr)

Reply:
Oh yeah it's just a small implementation detail. `rightEnumTypes` uses `.has` but the other doesn't. I didn't think it necessary to add more code or comments. 🤷